spec/controllers/answers_controller_spec.rb

```diff
@@ -121,52 +92,63 @@
 
       context 'invalid' do
         it 'not updated answer' do
-          patch :update, params: {id: answer, question_id: question, answer: attributes_for(:answer, :invalid_answer), user: user}
+          patch :update, params: {id: answer, question_id: question, answer: attributes_for(:answer, :invalid_answer), user: user, format: :js}
           answer.reload
 
           expect(answer.body).to eq 'MyText'
         end
 
         it 'render edit after fail update' do
-          patch :update, params: {id: answer, question_id: question, answer: attributes_for(:answer, :invalid_answer), user: user}
-          expect(response).to render_template(:edit)
+          patch :update, params: {id: answer, question_id: question, answer: attributes_for(:answer, :invalid_answer), user: user, format: :js}
+          expect(response).to render_template :update
         end
       end
     end
 
   end
 
+  describe 'PATCH #select_best' do
+    let!(:user) { create(:user) }
+    let!(:wrong_user) { create(:user) }
+    let!(:question) { create(:question, user: user) }
+    let!(:answer) { create(:answer, question: question, user: user) }
+
+    it 'author of question selected best answer' do
+      login(user)
+      patch :select_best, params: { id: answer }, format: :js
+      answer.reload
+
+      expect(answer.best).to be_truthy
```

Comment:
https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/predicate-matchers